src/main/java/jaci/openrio/toast/lib/registry/Registrar.java

```diff
@@ -32,6 +32,15 @@
      */
     public synchronized  <T extends Type> T fetch(ID id, Class<T> clazz, Supplier<T> creator) {
         Type in_register = registered.get(id);
+
+        if(in_register == null) {
+            for (ID key : registered.keySet()) {
+                if (key.equals(id)) {
+                    in_register = registered.get(key);
+                }
+            }
+        }
+
         if (in_register == null) {
```

Comment:
Could this be combined with the if() statement above it? Both are checking for a null